lib/std/build/OptionsStep.zig

```diff
@@ -114,12 +114,23 @@ pub fn addOption(self: *OptionsStep, comptime T: type, name: []const u8, value:
     }
     switch (@typeInfo(T)) {
         .Enum => |enum_info| {
-            out.print("pub const {} = enum {{\n", .{std.zig.fmtId(@typeName(T))}) catch unreachable;
+            const enum_type_name = blk: {
+                const type_name = @typeName(T);
+                const last_dot = std.mem.lastIndexOfScalar(u8, type_name, '.') orelse break :blk type_name;
```

Comment:
I don't have time right now to suggest an alternative but this is not an OK way to solve the problem

Reply:
It's true that I felt a bit dirty about this hack, but it was the only reasonable way I could think of that would keep a "clean" generated options file.

I've modified this PR to instead apply `std.zig.fmtId` in every place it might be required which also solves this issue, resulting option file:
```zig
pub const @".@build.src.shared.ZigVersion" = enum {
    master,
    @"0.7.0",
    @"0.7.1",
    @"0.8.0",
    @"0.8.1",
    @"0.9.0",
};
pub const data_version: @".@build.src.shared.ZigVersion" = @".@build.src.shared.ZigVersion".master;
pub const @"std.log.Level" = enum {
    err,
    warn,
    info,
    debug,
};
pub const log_level: @"std.log.Level" = @"std.log.Level".info;

```

Reply:
Just to note before #11226 the generated option file looked like:
```zig
pub const ZigVersion = enum {
    master,
    @"0.7.0",
    @"0.7.1",
    @"0.8.0",
    @"0.8.1",
    @"0.9.0",
};
pub const data_version: ZigVersion = ZigVersion.master;
pub const Level = enum {
    err,
    warn,
    info,
    debug,
};
pub const log_level: Level = Level.info;
```

Reply:
Thanks for making the changes and sorry for the unhelpful review earlier. Let's start with this fix and then see if we can come up with some follow-up improvements to it :+1: 

Before #11226 there was potential for completely unrelated types to clash names, and now that is solved, albeit with "uglier" identifier names. I think we can still do better, but let's think carefully about how to do that.